### PR TITLE
Linking relevant docs on `StreamingResponse`

### PR DESCRIPTION
**Pull request by jamesbraza**
In https://github.com/tiangolo/fastapi/discussions/10761, I asked how `StreamingResponse` works.

I discovered it is vendored by Starlette, so I have opened two PRs:
- This one, linking to Starlette docs
- https://github.com/encode/starlette/pull/2405 on how `StreamingResponse` matches the ASGI spec

### CODE REVIEW

**Comment by tiangolo** [comment]
📝 Docs preview for commit 19210fd65917faadf8eccdb9d58ba0c2f53405c4 at: https://b97be037.fastapitiangolo.pages.dev

File: docs/en/docs/advanced/custom-response.md [hunk 1]
```diff
@@ -202,6 +202,10 @@ Takes an async generator or a normal generator/iterator and streams the response
 {!../../../docs_src/custom_response/tutorial007.py!}
 ```
 
+For reference, `StreamingResponse` comes from [`starlette`](https://github.com/encode/starlette).
+More information on it can be found in Starlette's docs:
+https://www.starlette.io/responses/#streamingresponse.
```
Comment (hasansezertasan) [inline docs/en/docs/advanced/custom-response.md:207]:
```suggestion
You can read more details about the <a href="https://www.starlette.io/responses/#streamingresponse" class="external-link" target="_blank">`StreamingResponse` object in the official Starlette documentation site</a>.
```

**Review by hasansezertasan — CHANGES_REQUESTED** [review]
I have left a suggestion, could you please check it?

**Comment by hasansezertasan** [comment]
And I believe you should move that line to the end of the "### `StreamingResponse`" section.

File: docs/en/docs/advanced/custom-response.md (hunk 1 above)
Comment (jamesbraza) [reply docs/en/docs/advanced/custom-response.md:207]: Thank you for the suggestion, but I prefer to stick with Markdown built-in links, as opposed to new-tab opening. Most of the links in this doc do not use `target="_blank"` method. If others want a new tab, they can CTRL-click as is standard across the internet.

**Comment by jamesbraza** [comment]
I have moved this docs addition to its own section as suggested

**Comment by tiangolo** [comment]
📝 Docs preview for commit 044f9429fcd55a62cf0b6a7bfa57de0bd5d6deda at: https://a30737b2.fastapitiangolo.pages.dev

**Pull request closed without merging**